packages/utils/getSpacingStyle/src/getSpacingStyle.ts

```
@@ -0,0 +1,40 @@
+ * @description useSpacing is a library that makes the user's spacing style more efficient.
+ * @param { Object } options
+ * @param { number | "small" | "medium" | "large" } options.size
+ * @param { "all" | "horizontal" | "vertical" | "top" | "bottom" | "left" | "right" } options.direction
+ * @param { "px" | "rem" | "em" | "%" } [options.unit="px"]
+ * @returns { Record<string, string | number> }
+ */
+export const getSpacingStyle = ({ size, direction, unit = "px" }: SpacingOptions): Record<string, string | number> => {
+  const spacingValue = typeof size === "number" ? size : sizeMap[size];
+  const valueWithUnit = `${spacingValue}${unit}`;
+
+  return directionStyles[direction].reduce((acc, cur) => ({ ...acc, [cur]: valueWithUnit }), {});
```

Biome

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)